java/src/jmri/implementation/MatrixSignalMast.java

```diff
@@ -88,19 +88,20 @@ private void configureFromName(@Nonnull String systemName) {
         configureAspectTable(system, mast); // (create -default- appmapping in var "map") in AbstractSignalMast
     }
 
-    private HashMap<String, char[]> aspectToOutput = new HashMap<String, char[]>(16); // "Clear" - 01001 char[] pairs
+    private final HashMap<String, char[]> aspectToOutput = new HashMap<>(16); // "Clear" - 01001 char[] pairs
     private char[] unLitBits;
 
     /**
-     * Store bits in aspectToOutput hashmap.
+     * Store bits in aspectToOutput hashmap, synchronized.
+     * <p>
+     * Length of bitArray should match the number of outputs defined, so one digit per output.
      *
      * @param aspect String valid aspect to define
      * @param bitArray char[] of on/off outputs for the aspect, like "00010"
-     * Length of bitArray should match the number of outputs defined, so one digit per output
     */
-    public void setBitsForAspect(String aspect, char[] bitArray) {
+    public synchronized void setBitsForAspect(String aspect, char[] bitArray) {
         if (aspectToOutput.containsKey(aspect)) {
-            if (log.isDebugEnabled()) log.debug("Aspect {} is already defined as {}", aspect, java.util.Arrays.toString(aspectToOutput.get(aspect)));
+            log.debug("Aspect {} is already defined as {}", aspect, java.util.Arrays.toString(aspectToOutput.get(aspect)));
```

Comment:
In this case, `if (log.isDebugEnabled())` is desired since `java.util.Arrays.toString(aspectToOutput.get(aspect))` will otherwise be calculated even if debug is turned off.

Reply:
Rolled back as suggested, but notice there is now the if evaluation, and there's simple and complicated calculations.

Reply:
Can you explain that? I'm not sure I understand what you mean but I'm willing to learn.